knowledge-base/custom-definition-storage-does-not-implement-interface-member.md

```diff
@@ -0,0 +1,113 @@
+---
+title: CustomDefinitionStorage does not implement interface member IAssetsStorage.CreateFolderAsync(CreateFolderModel)
+description: The 'CustomDefinitionStorage' does not implement interface member 'IAssetsStorage.CreateFolderAsync(CreateFolderModel)' is thrown after upgrade to R2 2022
+type: troubleshooting
+page_title: 'CustomDefinitionStorage' does not implement interface member 'IAssetsStorage.CreateFolderAsync(CreateFolderModel)'
+slug: custom-definition-storage-does-not-implement-interface-member
+position: 
+tags: webreportdesigner
+ticketid: 1564724
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product Version</td>
+			<td>16.1.22.511+</td>
+		</tr>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+		<tr>
+			<td>Report Designer</td>
+			<td>Web Report Designer</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+The error `'CustomDefinitionStorage' does not implement interface member 'IAssetsStorage.CreateFolderAsync(CreateFolderModel)'` is thrown after upgrade to R2 2022.
+
+## Error Message
+'CustomDefinitionStorage' does not implement interface member 'IAssetsStorage.CreateFolderAsync(CreateFolderModel)'
+'CustomDefinitionStorage' does not implement interface member 'IAssetsStorage.DeleteAsync(string)'
+...
+
+
+## Cause\Possible Cause(s)
+In R2 2022, we improved the implementation of the IDefinitionStorage interface. Now the methods are asynchronous and you will need to replace
```

Comment:
put IDefinitionStorage  in code with `, or reference the interface from the API reference - https://docs.telerik.com/reporting/api/Telerik.WebReportDesigner.Services.IDefinitionStorage.html